src/core/core.datasetController.js

```diff
@@ -346,18 +346,41 @@ export default class DatasetController {
 
 		me._dataCheck();
 
-		const data = me._data;
-		const metaData = meta.data = new Array(data.length);
+		// const data = me._data;
+		// const metaData = meta.data = new Array(data.length);
 
-		for (let i = 0, ilen = data.length; i < ilen; ++i) {
-			metaData[i] = new me.dataElementType();
-		}
+		// for (let i = 0, ilen = data.length; i < ilen; ++i) {
+		// 	metaData[i] = new me.dataElementType();
+		// }
 
 		if (me.datasetElementType) {
 			meta.dataset = new me.datasetElementType();
 		}
 	}
 
+	_notifyAfterParse() {
+		const me = this;
+		const {_cachedMeta: meta} = me;
+		const {iScale} = meta;
+
+		// We expect afterDataParse hooks to set the modified data
```

Comment:
Maybe it'd be cleaner to always set `meta._parsed = meta._parsedRaw` and then have the plugin override `meta._parsed`. We could expose a `setParsed` method if that's preferable to setting the value of private `_parsed` variable